### ISSUE TITLE
processing 4-channel images

### ISSUE THREAD

**Issue by Pepslee** [alexgkendall/caffe-segnet#47]
Can SegNet process 4-channel ( or more channel ) images ?
What i need to change in source code, to process more then 3 channel images ?

**Comment by alexgkendall** [comment]
Yes absolutely. However PNG images in general cannot handle n-channel tensors. You will need to use another layer to input data. One option might be to arrange your data into an LMDB dataset and use the standard caffe data layer. Or you could modify the dense image data layer to support this.

**Comment by Pepslee** [comment]
Why can't I use 4-channel `.png` image, or 4-channel `.tif` image ?  Opencv `imread` can open 4-channel image, using `CV_LOAD_UNCHANGED` flag.

**Comment by alexgkendall** [comment]
Sure, you could do that. You'd need to make some changes to the dense_image_data_layer

**Comment by Pepslee** [comment]
I changeв the flag `CV_LOAD_COLOR` to `CV_LOAD_UNCHANGED`  in  `ReadImageToCVMat` function, but it did not work (

**Comment by leihamilton** [comment]
Hi, @alexgkendall, I would like to try RGBD data on SegNet, I followed your suggestions and created a lmdb database for RGBD, and a lmdb database for label. In the original SegNet, the data is first pasted into the NORM layer. My questions is that do I still pass the RGBD to the NORM layer? Thanks.

**Comment by Pepslee** [comment]
Hi, alexgkendall , you said, that to process 4-channel images, I need to make some changes to the dense_image_data_layer.  I made some changes to the **ReadImageToCVMat** function, to provide 4-channel image loading. But I have some [strenge result ](https://github.com/alexgkendall/caffe-segnet/issues/24), there are some vertical lines on the output mask.  These vertical lines appear every 4 pixels, ie problem for the 4-th channel.  Would I need to make some changes in the **upsample_layer** or in the **compute_bn_statistic_py** code ?????
